### PR TITLE
Add alert for failed requests

### LINKED ISSUE #1008: Improve error handling on fetching echo missions
**Describe the improvement you would like to see**
When fetching all echo missions with the `EchoController`, the status codes `502` or `500` is returned if an ecxeption is catched: 

```
try
        {
            var missions = await _echoService.GetAvailableMissions(plantCode);
            return Ok(missions);
        }
        catch (HttpRequestException e)
        {
            _logger.LogError(e, "Error retrieving missions from Echo");
            return new StatusCodeResult(StatusCodes.Status502BadGateway);
        }
        catch (JsonException e)
        {
            _logger.LogError(e, "Error retrieving missions from Echo");
            return new StatusCodeResult(StatusCodes.Status500InternalServerError);
        }
```

In the frontend, it will look like the request is spinning forever, even though the request does not repeat/retries. In this situation, it should instead display an error message to the user. 

**How will this change existing functionality?**
No change in functionality, just giving user meaningful error messages. 

**How will this improvement affect the current Threat Model?**
Should not affect threat model, but be aware of what to display in frontend

### PR DESCRIPTION
**Pull request by andchiind**
Closes #1008

### CODE REVIEW

**Comment by andchiind** [comment]
As a part of implementing the alert in such a way that the dialog closes itself when an error is returned, I ended up refactoring MissionQueueView.tsx and the dialog component file.

**Comment by Eddasol** [comment]
image
image
Gap between "Add mission" and "Create mission" changes when "Add mission" is pressed

File: frontend/src/components/Pages/FrontPage/MissionOverview/MissionQueueView.tsx [hunk 1]
```diff
@@ -43,55 +46,33 @@ export function MissionQueueView({ refreshInterval }: RefreshProps) {
...
         new Map<string, EchoMissionDefinition>()
     )
     const [robotOptions, setRobotOptions] = useState<Robot[]>([])
-    const [scheduleButtonDisabled, setScheduleButtonDisabled] = useState<boolean>(true)
     const [frontPageScheduleButtonDisabled, setFrontPageScheduleButtonDisabled] = useState<boolean>(true)
     const [isFetchingEchoMissions, setIsFetchingEchoMissions] = useState<boolean>(false)
+    const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false)
+    const [isAddMissionPopoverOpen, setIsAddMissionPopoverOpen] = useState<boolean>(false)
+    const anchorRef = useRef<HTMLButtonElement>(null)
     const { installationCode } = useInstallationContext()
```
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/MissionQueueView.tsx:59]: Set the contexts together at line 46 to 49 for better structure

**Comment by andchiind** [comment]
> image Suggest updating style in "Request error" to match existing style for failed mission

I've now made them as similar as possible without changing the information.

**Comment by andchiind** [comment]
> image image Gap between "Add mission" and "Create mission" changes when "Add mission" is pressed

This should be fixed now.

**Review by mrica-equinor — CHANGES_REQUESTED** [review]

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -5,27 +5,30 @@ import { LanguageProvider } from 'components/Contexts/LanguageContext'
...
-                                <FlotillaSite />
-                            </MissionFilterProvider>
-                        </AuthenticatedTemplate>
-                    </>
-                </MissionControlProvider>
-            </LanguageProvider>
-        </MissionsProvider>
+        <AlertProvider>
+            <MissionsProvider>
+                <LanguageProvider>
```
[the remark this answers is not in the document]
Comment (andchiind) [reply frontend/src/App.tsx:14]: The order matters in the sense that if you want to use a context, the component/context that uses it needs to be within the provider of the context you're using. So for MissionsProvider to use the alerts, AlertProvider needs to envelop it.

File: frontend/src/components/Alerts/FailedMissionAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+function FailedMission({ mission }: MissionProps) {
+    const { TranslateText } = useLanguageContext()
+    const navigate = useNavigate()
+    const goToMission = () => {
+        const path = `${config.FRONTEND_BASE_ROUTE}/mission/${mission.id}`
+        navigate(path)
+    }
+
+    return (
+        <Button as={Typography} onClick={goToMission} variant="ghost" color="secondary">
```
[the remark this answers is not in the document]
Comment (andchiind) [reply frontend/src/components/Alerts/FailedMissionAlert.tsx:30]: sorry but I don't suppose you remember the name of that colour?

[the remark this answers is not in the document]
Comment (andchiind) [reply frontend/src/components/Alerts/FailedMissionAlert.tsx:27]: I don't see it in EDS 
image

[the remark this answers is not in the document]
Comment (andchiind) [reply frontend/src/components/Alerts/FailedMissionAlert.tsx]: Done

**Review by mrica-equinor — APPROVED** [review]
LGTM :)

**Pull request merged**